assets/src/edit-story/components/panels/slug.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Row, TextInput, HelperText } from '../form';
+import { useStory } from '../../app/story';
+import { SimplePanel } from './panel';
+
+const Permalink = styled.a`
+  color: ${({ theme }) => theme.colors.link};
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px 6px;
+  border-radius: 4px;
+  flex-grow: 1;
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.fg.v1};
+  }
+`;
+
+function SlugPanel() {
+  const {
+    state: {
+      story: { slug, link },
+    },
+    actions: { updateStory },
+  } = useStory();
+  const handleChangeValue = useCallback(
```

Comment:
I would add some sanitation to the slug update similar this. 
https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/header/title.js#L67-L73

Reply:
Added `cleanForSlug`. 

@swissspidy Any reason for `cleanForSlug` removing `-` from the value? Is that purposefully not allowed?